### PR TITLE
AzureAD Applications (Independent Publisher) v 1.01

### PR DESCRIPTION
**Pull request by paulculmsee**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [ x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [ x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x ] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ x] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
-
image
image
image
image

- [x ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- 
image

- [ x] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

File: independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json [hunk 1]
```diff
@@ -49,23 +54,60 @@
...
             "name": "$count",
             "in": "query",
             "required": false,
             "type": "string",
             "description": "true or false - Retrieves the total count of matching resources",
             "x-ms-summary": "Display count",
-            "x-ms-visibility": "advanced"
+            "x-ms-visibility": "advanced",
+			"default" : "true",
+			"enum": [
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json:89]: @paulculmsee The list of enum values has been added to the input field. Existing flows may use values outside of this list. can you please confirm that no users will be affected by this change?

File: independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json [hunk 2]
```diff
@@ -81,405 +123,519 @@
...
+			"description": "The groups claim issued in a user or OAuth 2.0 access token that the application expects"
+		  },
+		  "identifierUris": {
+			"type": "array",
+			"items": {
+			  "type": "string"
+			},
+			"description": "The URIs that identify the application within its Azure AD tenant, or within a verified custom domain if the application is multi-tenant"
+		  },
+		  "isDeviceOnlyAuthSupported": {
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json:281]: @paulculmsee The type has been changed for the output field. Existing flows may be broken if they are using this field and actual payload has been changed. can you please define the "type'. Or please confirm that no users will be affected by this change.

File: independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json [hunk 3]
```diff
@@ -81,405 +123,519 @@
...
+		  },
+		  "isFallbackPublicClient": {
+			"type": "boolean",
+			"description": "Specifies the fallback application type as public client, such as an installed application running on a mobile device. The default value is false which means the fallback application type is confidential client such as a web app"
+		  },
+		  "notes": {
+			"type": "string",
+			"description": "Notes relevant for the management of the application"
+		  },
+		  "optionalClaims": {
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json:292]: @paulculmsee The type has been changed for the output field. Existing flows may be broken if they are using this field and actual payload has been changed. can you please define the "type'. Or please confirm that no users will be affected by this change.

File: independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json [hunk 4]
```diff
@@ -81,405 +123,519 @@
...
+				}
+			  }
+			},
+			"description": "Defines custom behavior that a consuming service can use to call an app in specific contexts"
+		  },
+		  "api": {
+			"type": "object",
+			"properties": {
+			  "acceptMappedClaims": {
+				"type": "boolean",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json:372]: @paulculmsee The type has been changed for the output field. Existing flows may be broken if they are using this field and actual payload has been changed. can you please confirm that no users will be affected by this change?

File: independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json [hunk 5]
```diff
@@ -49,23 +54,60 @@
...
+            "name": "ConsistencyLevel",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "description": "Set this property to 'eventual' if you are using $search",
+            "x-ms-summary": "Enable advanced queries",
+            "x-ms-visibility": "advanced",
+			"enum": [
+              "eventual",
+              ""
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json:76]: @paulculmsee is it possible to remove the empty string?

File: independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json [hunk 6]
```diff
@@ -49,23 +54,60 @@
...
+          },
+          {
+            "name": "ConsistencyLevel",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "description": "Set this property to 'eventual' if you are using $search",
+            "x-ms-summary": "Enable advanced queries",
+            "x-ms-visibility": "advanced",
+			"enum": [
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json:74]: @paulculmsee all files should be properly formatted. may I ask you please to fix the indentation for all the files.

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

File: independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json (hunk 1 above)
Comment (paulculmsee) [reply independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json:89]: I don't believe there will be an impact as users were not having success with the connector and there are only two possible values. However, I just noticed that I left in a policy template that was setting this header so I need to remove it and retest. So I will push an update out tomorrow morning

File: independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json (hunk 5 above)
Comment (paulculmsee) [reply independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json:76]: There is only 1 value for that property - it is there or it is not. I could not see a better way to do it. Thoughts @Amjed-Ayoub ?

File: independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json (hunk 1 above)
Comment (paulculmsee) [reply independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json]: Left policy template in and retested. Relates to ms graph API advanced search queries. This enum has been removed altogether

File: independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json (hunk 2 above)
Comment (paulculmsee) [reply independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json]: I could not pinpoint what you are referring to here sorry. I abstracted the schema into a separate definition for all actions, and there was a couple of erroneous ones that only manifested at scale

File: independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json (hunk 5 above)
Comment (paulculmsee) [reply independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json]: Removed property entirely and went with a policy template

File: independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json (hunk 2 above)
Comment (jopanchal) [reply independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json]: @Amjed-Ayoub  are you referring to schema changes that paul made to correct the original errors, or what sections specifically? These changes were fixes.

Comment (Amjed-Ayoub) [reply independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json]: Hello @paulculmsee,

this property used to have a type of string. please see below
image

In this update there is no type defined.

File: independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json (hunk 3 above)
Comment (Amjed-Ayoub) [reply independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json]: Hello @paulculmsee,

this property used to have a type of string. please see below
image

In this update there is no type defined.

File: independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json (hunk 4 above)
Comment (Amjed-Ayoub) [reply independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json]: @paulculmsee 
the type for this property used to be as string, but now it got changed to boolean.
image

Comment (paulculmsee) [reply independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json]: Ah yes - I tested the connector on a reasonable sized tenant with some 1700+ registrations and realised that it was a schema error not picked up on v1. I'll run another test to verify for you to confirm.

Comment (paulculmsee) [reply independent-publisher-connectors/AzureADApplications/apiDefinition.swagger.json]: Confirmed, it should have been boolean. 
image

**Comment by Amjed-Ayoub** [comment]
@paulculmsee May I ask you please to resolve all the above conversations? Thank you so much.

File: independent-publisher-connectors/AzureADApplications/apiProperties.json [hunk 7]
```diff
@@ -5,9 +5,10 @@
         "type": "oauthSetting",
         "oAuthSettings": {
           "identityProvider": "aad",
-          "clientId": "[[DUMMY]]",
+          "clientId": "fbe18248-e189-46d5-a4bf-b1f1b854a48c",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/AzureADApplications/apiProperties.json:8]: @paulculmsee it is not allowed to have secret look alike values. Please replace it with a dummy value.

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

File: independent-publisher-connectors/AzureADApplications/apiProperties.json (hunk 7 above)
Comment (paulculmsee) [reply independent-publisher-connectors/AzureADApplications/apiProperties.json]: Apologies - should have caught that one

**Comment by Amjed-Ayoub** [comment]
[[certify-connector]]

**Comment by Amjed-Ayoub** [comment]
Hello @paulculmsee,
I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks. 
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by Amjed-Ayoub** [comment]
Hello @paulculmsee,
I hope you are doing well.
Your connector is in production deployment and queued up for the next deployment schedule. This process typically takes 3-4 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps.
Please let us know if you have any questions.
Thank you very much for working with us.

**Review by Amjed-Ayoub — APPROVED** [review]
Hello @paulculmsee ,
I hope you are doing well.
Congratulations, the deployment of your connector and documentation page was already completed, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program.
Thank you very much for working with us.

**Pull request merged**